reveal/run.js

```diff
@@ -88,8 +91,23 @@
     });
   }
 
-  function hasRemoteQueryParameter() {
-    return window.location.search.match(/[?&]remote[&]?/i);
+  function selectWidth(queryParams) {
+    if (queryParameterIsPresent(window.location.search, 'wide')) return 1413; // 16:9
+    else if (queryParameterIsPresent(window.location.search, 'wide10')) return 1272; // 16:10
+    else return 1124; // ~4:3
+  }
+
+  function cycleWideModes() {
+    var queryString = window.location.search;
```

Comment:
That what I did initially, but it does not work if the code needs to make 2 modifications inside the same event loop turn, which is what I need to do in `cycleWideModes` (remove `wide` then add `wide10`). That is because `window.location.search` is not updated until the page refreshes. Try it out with `console.log(window.location.search); window.location.search = newValue; console.log(window.location.search);`. Both logs show the same value. Another solution to this problem could have been to write a new method `replaceQueryParameter` but I thought making all the method pure would have the advantage of making the code clearer.